patches/tModLoader/Terraria/GameContent/ShopHelper.cs.patch

```diff
@@ -32,3 +32,13 @@
  		}
  
  		private void AddHappinessReportText(string textKeyInCategory, object substitutes = null) {
++			string str;
++			if (_currentNPCBeingTalkedTo.type <= NPCID.Count)
+-			string str = "TownNPCMood_" + NPCID.Search.GetName(_currentNPCBeingTalkedTo.netID);
++				str = "TownNPCMood_" + NPCID.Search.GetName(_currentNPCBeingTalkedTo.netID);
++			else
++				str = "Mods." + _currentNPCBeingTalkedTo.ModNPC.Mod.Name + "." + _currentNPCBeingTalkedTo.ModNPC.Name + "." + "TownNPCMood";
++
```

Comment:
Cache search result, use interpolated strings, using a local for modnpc would also be nice.

Reply:
Resolved exempting caching search result, not understood what to do with that.
Should be sufficient now